docs/dev/framework/security.md

```diff
@@ -0,0 +1,275 @@
+```
+
+{{% notice tip %}}
+Since Contao **4.10** there are class constants available for the various permission attributes, so that you do not have to remember them
+yourself and instead can use your IDE to find the correct attribute. For the Contao Core these constants are available in 
+`Contao\CoreBundle\Security\ContaoCorePermissions` while the permissions of the additional bundles are available in 
+`Contao\NewsBundle\Security\ContaoNewsPermissions`, `Contao\CalendarBundle\Security\ContaoCalendarPermissions`, 
+`Contao\NewsletterBundle\Security\ContaoNewsletterPermissions` and `Contao\FaqBundle\Security\ContaoFaqPermissions` respectively. The
+PHP documentation of these constants also contain hints about what type the subject should be. Examples:
+
+```php
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Contao\NewsBundle\Security\ContaoNewsPermissions;
+
+// Whether user can access the news module in the back end
+$security->isGranted(ContaoCorePermissions::USER_CAN_ACCESS_MODULE, 'news');
+
+// Whether user can use the hidden input field in the form generator
+$security->isGranted(ContaoCorePermissions::USER_CAN_ACCESS_FIELD, 'hidden');
+
+// Whether user has access to any field of tl_content
+$security->isGranted(ContaoCorePermissions::USER_CAN_EDIT_FIELDS_OF_TABLE, 'tl_content');
+
+// Whether user can create news archives
+$security->isGranted(ContaoNewsPermissions::USER_CAN_CREATE_ARCHIVES);
+```
+{{% /notice %}}
+
+
+## Custom Back End Access Rights
+
+To implement your own back end access rights (e.g. for custom modules in the back end) the following steps are necessary:
+
+1. Add your new permission to `$GLOBALS['TL_PERMISSIONS']`.
+2. Add your new permission to `tl_user` and `tl_user_group`.
+
+```php
+// contao/config/config.php
+$GLOBALS['TL_PERMISSIONS'][] = 'my_permission';
+```
+
+```php
+// contao/dca/tl_user.php
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+
+$GLOBALS['TL_DCA']['tl_user_group']['fields']['my_permission'] = [
+    'exclude' => true,
+    'inputType' => 'checkbox',
+    'sql' => ['type' => 'boolean', 'default' => false],
+];
+
+PaletteManipulator::create()
+    ->addLegend('my_legend', null)
+    ->addField('my_permission', 'my_legend', PaletteManipulator::POSITION_APPEND)
+    ->applyToPalette('extend', 'tl_user')
+    ->applyToPalette('custom', 'tl_user')
+;
+```
+
+```php
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+
+$GLOBALS['TL_DCA']['tl_user_group']['fields']['my_permission'] = [
+    'exclude' => true,
+    'inputType' => 'checkbox',
+    'sql' => ['type' => 'boolean', 'default' => false],
+];
+
+PaletteManipulator::create()
+    ->addLegend('my_legend', null)
+    ->addField('my_permission', 'my_legend', PaletteManipulator::POSITION_APPEND)
+    ->applyToPalette('default', 'tl_user_group')
+;
+```
+
+Once that is done you can check for this permission in your own back end controller for example. These permissions can be checked via the 
```

Comment:
unfortunately, if I remember correctly, you cannot have a single boolean field as a permission (inheritance (`TL_PERMISSION`) doesnt work with that). It has to be a multiple-checkbox that can be `array_merge`d

Reply:
It does work as far as I tested (see the example below).

Reply:
I don't think it does 😉

Reply:
Not sure it does. The result is always an array for sure, which looks pretty confusing:
https://github.com/contao/contao/blob/4.x/core-bundle/src/Resources/contao/classes/BackendUser.php#L446-L447

Reply:
Yes, and this results in the following check in `hasAccess`: https://3v4l.org/jhoYa

And thus it will return true.

Reply:
it _might_ work, yes. I remember I had issues with this, probably in Contao 2 or 3. It will however add confusing data, since the user object will now hold an array of this field, even though the user field is a checkbox only.
I'm just saying, maybe it's better to make an example of what we use everywhere (multiple checkboxes) instead of just one 😊